ui/app/src/components/WorkflowStateManagement/WorkflowStateManagement.tsx

```diff
@@ -123,6 +127,12 @@ export function WorkflowStateManagement(props: WorkflowStateManagementProps) {
     fetchStates();
   }, [fetchStates]);
 
+  useEffect(() => {
+    fnRefs.current.onSubmittingAndOrPendingChange?.({
+      hasPendingChanges: Boolean(selectedItemsLength)
+    });
+  }, [selectedItemsLength, fnRefs]);
```

Comment:
`selectedItemsLength` may change too much. All we would care about is when it changes from or to `0`. Doesn't `hasSelectedItems` give you that already?